app/Helpers/database/ticket.php

```diff
@@ -15,35 +15,36 @@
 use Illuminate\Support\Facades\Cache;
 use Illuminate\Support\Facades\DB;
 
-function isAllowedToSubmitTickets(string $user): bool
+function isAllowedToSubmitTickets(string $username): bool
 {
-    if (!isValidUsername($user)) {
+    if (!isValidUsername($username)) {
         return false;
     }
 
-    $cacheKey = CacheKey::buildUserCanTicketCacheKey($user);
+    $cacheKey = CacheKey::buildUserCanTicketCacheKey($username);
 
-    $value = Cache::get($cacheKey);
-    if ($value !== null) {
-        return $value;
+    if (Cache::has($cacheKey)) {
+        return Cache::get($cacheKey);
```

Comment:
Doesn't this cause a double lookup?

Reply:
Without the join against Activity, we could probably just eliminate caching of this altogether. What about something like this?
```
function isAllowedToSubmitTickets(string $username): bool
{
    $user = User::firstWhere('User', $username);
    if (!$user || $user->Created->diffInDays() < 1) { 
        return false;
    }

    return PlayerGame::where('user_id', $user->id)
        ->where('time_taken', '>', 5)
        ->exists();
}
```

Reply:
very nice - changed in https://github.com/RetroAchievements/RAWeb/pull/1954/commits/8bbe110fd76f6fabf42f8596df0c39d3a09e1186

Reply:
Should also remove the code for `buildUserCanTicketCacheKey`

Reply:
removed in https://github.com/RetroAchievements/RAWeb/pull/1954/commits/ea238e5d0169684a295cd71f79b748907b76a9c9